src/overlays/actors/ovl_Boss_Ganondrof/z_boss_ganondrof.c

```diff
@@ -61,71 +148,1395 @@ static ColliderCylinderInit D_80914CFC = {
+};
+static UNK_PTR D_80915074_16x32[] = { &gPhantomGanonUnknown_00AA80, &gPhantomGanonUnknown_00AF80 };
+static UNK_PTR D_8091507C_16x16[] = { &gPhantomGanonUnknown_0040B0, &gPhantomGanonUnknown_003FB0 };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_U8(targetMode, 5, ICHAIN_CONTINUE),
+    ICHAIN_S8(naviEnemyId, 43, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, 0, ICHAIN_CONTINUE),
+    ICHAIN_F32(targetArrowOffset, 0, ICHAIN_STOP),
+};
+
+void BossGanondrof_ClearPixels8x8(s16* texture, u8* mask, s16 index) {
+    if (mask[index]) {
+        texture[index / 4] = 0;
+    }
+}
+
+void BossGanondrof_ClearPixels16x8(s16* texture, u8* mask, s16 index) {
+    if (mask[index]) {
+        texture[index / 2] = 0;
+    }
+}
+
+void BossGanondrof_ClearPixels16x16(s16* texture, u8* mask, s16 index) {
+    if (mask[index]) {
+        texture[index] = 0;
+    }
+}
+
+void BossGanondrof_ClearPixels32x16(s16* texture, u8* mask, s16 index) {
+    s16 tmp;
+
+    if (mask[index]) {
+        tmp = (index & 0xF) + ((index & 0xF0) << 1);
+        texture[tmp + 0x10] = 0;
+        texture[tmp] = 0;
+    }
+}
+
+void BossGanondrof_ClearPixels16x32(s16* texture, u8* mask, s16 index) {
+    s16 tmp;
+
+    if (mask[index]) {
+        tmp = ((index & 0xF) * 2) + ((index & 0xF0) * 2);
+        texture[tmp + 1] = 0;
+        texture[tmp] = 0;
+    }
+}
+
+void BossGanondrof_ClearPixels(u8* mask, s16 index) {
+    s16 i;
+
+    for (i = 0; i < 5; i++) {
+        BossGanondrof_ClearPixels8x8(SEGMENTED_TO_VIRTUAL(D_80915028_8x8[i]), mask, index);
+        BossGanondrof_ClearPixels16x8(SEGMENTED_TO_VIRTUAL(D_8091503C_16x8[i]), mask, index);
+    }
+
+    for (i = 0; i < 9; i++) {
+        BossGanondrof_ClearPixels16x16(SEGMENTED_TO_VIRTUAL(D_80915050_16x16[i]), mask, index);
+    }
+
+    for (i = 0; i < 2; i++) {
+        BossGanondrof_ClearPixels16x32(SEGMENTED_TO_VIRTUAL(D_80915074_16x32[i]), mask, index);
+    }
+
+    BossGanondrof_ClearPixels32x16(SEGMENTED_TO_VIRTUAL(&gPhantomGanonUnknown_00B380), mask, index);
+    BossGanondrof_ClearPixels16x32(SEGMENTED_TO_VIRTUAL(&gPhantomGanonUnknown_003DB0), mask, index);
+    for (i = 0; i < 2; i++) {
+        BossGanondrof_ClearPixels16x16(SEGMENTED_TO_VIRTUAL(D_8091507C_16x16[i]), mask, index);
+    }
+}
+
+void BossGanondrof_SetColliderPos(Vec3f* pos, ColliderCylinder* collider) {
+    collider->dim.pos.x = pos->x;
+    collider->dim.pos.y = pos->y;
+    collider->dim.pos.z = pos->z;
+}
+
+void BossGanondrof_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BossGanondrof* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    Actor_SetScale(&this->actor, 0.01f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &gPhantomGanonSkel_00C710, &gPhantomGanonAnim_003CA4, NULL, NULL, 0);
+    if (this->actor.params < 10) {
+        this->actor.params = 1;
```

Comment:
think you have params enum? dont remember

Reply:
same situation as the horse. needs a #define. I'll figure out how to handle it.